File: LeetCode Problems/301 to 400/343. Integer Break/343. Integer Break.js
/**
 * @param {number} n
 * @return {number}
 */
var integerBreak = function(n) {
    if (n <= 3)
        return n - 1
    var ans = 1
    while (n > 4) {
        n -= 3
        ans *= 3
    }
    return ans * n
};

File: LeetCode Problems/301 to 400/343. Integer Break/343. Integer Break.kt
class Solution {
    fun integerBreak(n: Int): Int {
        if (n <= 3)
            return n - 1;
        var ans = 1;
        var num = n;
        while (num > 4) {
            ans *= 3;
            num -= 3;
        }
        return ans * num;
    }
}